astropy/modeling/physical_models.py

```diff
@@ -371,3 +372,412 @@ def input_units(self):
+
+    Parameters
+    ----------
+    x_0 : float or :class:`~astropy.units.Quantity`
+        Radial position of the peak (center) of the NFW profile.
+    mass : float or class:`~astropy.units.Quantity`
+        Mass of NFW peak within specified overdensity radius.
+    concentration : float
+        Concentration of the NFW profile.
+    redshift : float
+        Redshift of the NFW profile.
+
+
+    Notes
+    -----
+
+    Model formula:
+
+    .. math:: \rho(r)=\\frac{\\delta_c\\rho_{c}}{r/r_s(1+r/r_s)^2}
+
+    References
+    ----------
+    .. [1] https://arxiv.org/pdf/astro-ph/9508025
+    .. [2] https://en.wikipedia.org/wiki/Navarro%E2%80%93Frenk%E2%80%93White_profile
+    """
+
+    # Model Parameters
+
+    # NFW Peak center
+    x_0 = Parameter(default=0, unit=u.Mpc)
+
+    # NFW Profile mass and mass type
+    mass = Parameter(default=1.0, min=1.0, unit=u.M_sun)
+
+    # NFW profile concentration
+    concentration = Parameter(default=1.0, min=1.0)
+
+    # NFW Profile redshift
+    redshift = Parameter(default=0.0, min=0.0)
+
+    # We allow values without units to be passed when evaluating the model, and
+    # in this case the input x values are assumed to be lengths / positions in Mpc.
+    _input_units_allow_dimensionless = True
+
+    def __init__(self, x_0=u.Quantity(x_0.default, x_0.unit),
+                 mass=u.Quantity(mass.default, mass.unit),
+                 concentration=concentration.default, redshift=redshift.default, massfactor="200c",
+                 cosmo=None,  **kwargs):
+        """
+        NFW profile Initialization
+
+        Parameters
+        ----------
+        x : float or :class:`~astropy.units.Quantity`
+            Radial position of density to be calculated for the NFW profile.
+        x_0 : float or :class:`~astropy.units.Quantity`
+            Radial position of the peak (center) of the NFW profile.
+        mass : float or class:`~astropy.units.Quantity`
+            Mass of NFW peak within specified overdensity radius.
+        concentration : float
+            Concentration of the NFW profile.
+        redshift : float
+            Redshift of the NFW profile.
+        massfactor : str
+            Mass overdensity factor and type for provided profiles:
+                vir : virial radius
+                Nc  : radius where density is N times that of the critical density (e.g. '200c')
+                Nm  : radius where density is N times that of the mean density (e.g. '500m')
+        cosmo : class:`~astropy.cosmology`
+            Background cosmology for density calculation.
+
+        """
+        # Set default cosmology
+        if cosmo is None:
+            cosmo = cosmology.default_cosmology.get()
+
+        # Set mass overdensity type and factor:
+        #   vir : virial radius
+        #   #c  : radius where density is # that of the critical density
+        #   #m  : radius where density is # that of the mean density
+        if massfactor.lower() == "vir":
```

Comment:
Would it be better to have a more descriptive string here?  `NFW1D(massfactor="virial")` would be clearer.